src/agent.cpp

```diff
@@ -198,12 +203,45 @@ JNIEXPORT jobjectArray JNICALL Java_com_intellij_memory_agent_proxy_IdeaNativeAg
 }
 
 extern "C"
-JNIEXPORT jobjectArray JNICALL Java_com_intellij_memory_agent_proxy_IdeaNativeAgentProxy_getAllReachableObjects(
+JNIEXPORT jobjectArray JNICALL Java_com_intellij_memory_agent_IdeaNativeAgentProxy_getAllReachableObjects(
         JNIEnv *env,
         jobject thisObject,
         jobject startObject,
         jobject suspectClass) {
     return GetAllReachableObjectsOfClassAction(env, gdata->jvmti, thisObject).run(startObject, suspectClass);
 }
 
+extern "C"
+JNIEXPORT jint JNICALL Java_com_intellij_memory_agent_IdeaNativeAgentProxy_addAllocationListener(
+        JNIEnv *env,
+        jclass thisClass, 
+        jobject listener,
+        jobjectArray trackedClasses) {
+    if (!canSampleAllocations) {
```

Comment:
Shouldn't we raise an exception if sampling isn't available?

Reply:
It is raised on the Java side